src/main/scala/Utilities.scala

```diff
@@ -0,0 +1,40 @@
+import org.apache.log4j.Level
+import java.util.regex.Pattern
+import java.util.regex.Matcher
+
+object Utilities {
+  /** Makes sure only ERROR messages get logged to avoid log spam. */
+  def setupLogging() = {
+    import org.apache.log4j.{Level, Logger}
+    val rootLogger = Logger.getRootLogger()
+    rootLogger.setLevel(Level.ERROR)
+  }
+
+  /** Configures Twitter service credentials using twiter.txt in the main workspace directory */
+  def setupTwitter() = {
+    import scala.io.Source
+
+    for (line <- Source.fromFile("resources/creds.txt").getLines) {
```

Comment:
the resource (`creds.txt`)should be present on classpath in order to get inside of final jar artifact after build.
see the [short explanation ](https://stackoverflow.com/a/27361820) 

and read about how class loading works and about ClassNotFoundException / NoClassDefFoundError  issues (they highlight common issues with spark dependencies).